File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n=nums.length
   // let set=new Set(nums)
    nums.sort((a,b)=>a-b);
    for(let i=0;i<n;i++){
        if(nums[i]!==i){
            return i
        }
    }
    return n
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n=nums.length;
        HashSet<Integer> seen =new HashSet<>();
        for(int i=0;i<n;i++){
            seen.add(nums[i]);
        }
        for(int i=0;i<=n;i++){
            if(!seen.contains(i)){
                return i;          }
        }
        return -1;
    }
}